app/models/rating.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+class Rating < ActiveRecord::Base
+  belongs_to :rateable, polymorphic: true, autosave: true
+  has_many :downvotes
+  has_many :upvotes
+  has_many :downvoters, through: :downvotes, source: :user, foreign_key: 'rater_id'
+  has_many :upvoters, through: :upvotes, source: :user, foreign_key: 'rater_id'
+
+  def increase_by(user:)
+    if decreased_by?(user)
+      self.value += 2
+      downvoters.delete(user)
```

Comment:
Is it possible for `downvoters.delete(user)` to succeed, but `upvoters << user` later to fail? What will be in the `value` variable in such a case? How to avoid this possible data inconsistency?